src/lib/krb5/os/locate_kdc.c

```diff
@@ -506,10 +535,188 @@ prof_locate_server(krb5_context context, const krb5_data *realm,
 }
 
 #ifdef KRB5_DNS_LOOKUP
+
+/*
+ * Parse the initial part of the URI, first confirming the scheme name.  Get
+ * the transport, master status, and host.  The host is either an address or
+ * hostname with an optional port, or an HTTPS URL.
+ * The format is krb5srv:[M]:udp|tcp|kkdcp:host
+ *
+ * Return a NULL *host_out if there's any problems parsing the URI.
+ */
+static void
+parse_uri_fields(const char *uri, k5_transport *transport_out,
+                 const char **host_out, int *master_out)
+
+{
+    k5_transport transport;
+    int master = FALSE;
+
+    *transport_out = 0;
+    *host_out = NULL;
+    *master_out = -1;
+
+    /* Confirm the scheme name. */
+    if (uri == NULL || (strncasecmp(uri, "krb5srv", 7) != 0))
```

Comment:
uri can't be NULL here (looking at the caller), so don't check.  (API functions can be paranoid since they expect an unbounded number of callers; helper functions should be strict and concise so that contracts are clear.)